lib/broadway_rabbitmq/amqp_client.ex

```diff
@@ -301,6 +310,16 @@ defmodule BroadwayRabbitMQ.AmqpClient do
     Connection.close(channel.conn)
   end
 
+  defp open_connection_instrumented(%{connection: connection} = config) when is_binary(connection) do
```

Comment:
Do we need this separate branch? Also, could we pass all options forward, instead of only `ssl_options`?

Reply:
This function can be a bit confusing, but what I am seeing is that `AMQP.Connection.open` can receive a URL or Keyword with host, username, password and etc.

So this branch is to make it work two scenarios:

1. Url is string, so we call `AMQP.Connection.open/2` passing url and options
2. connection is a Keyword, so `AMQP.Connection.open/1` is called.

Maybe for scenario 2 I can append the ssl options, but I see it already expects it as part of the possible options schema

Reply:
In other words, ssl_options are already supported if you pass a keyword list? Maybe you should pass a keyword list then and document this behaviour?

Reply:
In theory yes, I will test that to make sure It works as I am expecting it to.